index.html

```diff
@@ -2433,21 +2455,55 @@ <h2>
           <ol class="algorithm">
             <li>Set the [=assigned launch consumer=] to |consumer|.
             </li>
-            <li>If the [=unconsumed launch params=] on the {{Window/window}} is
-            set:
+            <li>[=list/For each=] [=unconsumed launch params=]:
+            </li>
+            <li>
+              <ol>
+          <li>Call |consumer| with [=unconsumed launch params=].
+          </li>
+              </ol>
+            </li>
+            <li>Clear the list of [=unconsumed launch params=].
+            </li>
+          </ol>
+        </section>
+      </section>
+      <section>
+        <h3>
+          Handling Web App Launches
+        </h3>
+        <p>
+          The steps to <dfn>launch a web app</dfn> are given by the 
+          following algorithm. The algorithm takes {{LaunchParams}}
+          |params:LaunchParams|.
+        </p>
+        <ol class="algorithm">
+          <li>If |params:LaunchParams| is null, set |params:LaunchParams| to
+            a new {{LaunchParams}} with params.targetURL set to
+            [=manifest/start_url=].
+          </li>
+          <li>Append |params| to the [=unconsumed launch params=] of the 
+            launched document's |window.launchQueue|.
+          </li>
+          <li>If the [=assigned launch consumer=] |consumer| is set:
+          </li>
+          <li>
```

Comment:
seems like there's an extra li and ol here?

Reply:
Done